articles/imbalanced-data-in-ml/index.md

```diff
@@ -0,0 +1,126 @@
+
+To elaborate further, it finds the [k-nearest-neighbors](https://deepai.org/machine-learning-glossary-and-terms/kNN#:~:text=The%20k-nearest%20neighbors%20algorithm%2C%20or%20kNN%2C%20is%20one,the%20longer%20it%20takes%20to%20perform%20the%20classification.) (k-NN) in the minority class. The k-NN refers to a classification approach where the likelihood of a data point belongs to one group or another depending on the data points in closest proximity to the data point.
+
+SMOTE chooses an instance of the minority class at random and computes its k-NN. A neighbor to it is then chosen randomly. After that, a synthetic example is created at a point selected at random between the two examples. This process can generate as many synthetic examples as would be needed for a minority class to create balance.
+
+A benefit of oversampling is that there is no data loss from the original training set. All the data from both the majority and minority classes are used. Yet, the downside of oversampling is that it causes overfitting.
+
+#### Undersampling
+
+When there exists a class that is in abundance, underfitting aims to reduce the size of the abundant class to balance the dataset. Using a similar context to the oversampling example, we have a classification problem with two classes and 100K data points. 20K data points are of the positive class, 80K for the negative class. We would need to undersample the majority class. This would involve choosing 20K data points randomly from the 80K available. We will then have 20K positive and 20K negative data points, bringing the total dataset size to 40K data points.
+
+For classification problems, there exists a method known as [Tomek links](http://www.samdrazin.com/classes/een548/project1report.pdf). It aims to improve the accuracy of the classification of data. This is by removing as much class label noise as possible. It also works to remove borderline examples with a higher probability of being incorrect. This is known as Tomek link removal. Tomek links are points that are the closest neighbors to each other but have different class labels. Since this technique makes it possible to identify points near different class labels, it is easy to get rid of them until none are left. In the context of undersampling, this technique gets rid of unwanted overlap between classes. The result is having only neighbors of the same class in close proximity.
```

Comment:
Can you define class label noise? How is it different from support vector machines? Even SVMs try to maximize the distance between the decision boundary and the support vectors.

Reply:
If similar, can support vectors act as tomek links? If different how?